azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java

```diff
@@ -1131,6 +1142,56 @@ private void createPod(final int executionId)
     if (flowParam != null && !flowParam.isEmpty()) {
       logger.info("ExecId: {}, Flow Parameters are: {}", executionId, flowParam);
     }
+
+    /*
+     The Set proxyUsersMap is being created with the intention of determining all the proxy users
+     prior to pod dispatch.
+     Why this is needed ?
+     -> When we use POLP(Principle of Least Priviledge) , initially introduced in
+     https://github.com/azkaban/azkaban/pull/3216, we need to have a list of users that custom
+     credentials (getCustomCredentialProvider) need to be fetched and registered for, as we will
+     not be able to fetch them at runtime as Azkaban user present in the Azkaban Executor Pod
+     will no longer be able to do so. This prefetching of custom credentials  is performed in
+     the azkaban-security-init container by attaching the proxy users needed to be prefetched as
+     environment variables.
+     */
+    Set<String> proxyUsersMap = new HashSet<>();
+
+    final boolean prefetchAllProxyUserCredentials =
+        flowParam != null && flowParam.containsKey(FlowParameters.PROXY_USER_PREFETCH_ALL) &&
+            Boolean.parseBoolean(flowParam.get(FlowParameters.PROXY_USER_PREFETCH_ALL));
+    /*
+     As an optimization for not walking through the DAG and making DB queries for large DAGs we
+     try to check if the proxy users list from the project page is less then a certain threshold.
+     If it is, less than this threshold, we simply add all of them. This configuration will be
+     custom to a given environment. An option to fetch all of them via the project page is
+     also provided with a flow parameter.
+     */
+
+    if (flow.getProxyUsers().size() > this.proxyUserPrefetchThreshold || prefetchAllProxyUserCredentials) {
```

Comment:
The comment above indicates that `prefetchAllProxyUserCredentials=true` means 'prefetch all proxy users listed in the project page', however the code says if true to traverse the DAG to gather proxy users for current execution. Could you please clarify what's the expected behavior?

Reply:
sorry, i think i removed the not condition. Adding it back .